etna/records/templatetags/records_tags.py

```diff
@@ -21,33 +23,60 @@ def record_url(
     Handling of Iaid as priority to allow Iaid in disambiguation pages when
     returning more than one record
     """
-    if iaid := record.get("iaid"):
+    try:
+        # Use Record property if available
+        iaid = record.iaid
+    except AttributeError:
+        # 'record' is likely just a dict
+        iaid = record.get("iaid")
+    except (ValueExtractionError, ValueError):
```

Comment:
`ValueError` is a particular case for exception handling. 
`ValueError` cases where `iaid` are not yet brought into Etna i.e. does not match catalogue `IAIDConverter` regex - example iaid for related article for blog post.
It would be nice to have some comment on it.